nixos/modules/services/web-apps/tachidesk-server.nix

```diff
@@ -0,0 +1,211 @@
+
+  config = mkIf cfg.enable {
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.config.port ];
+
+    users.groups = mkIf (cfg.group == "tachidesk") {
+      tachidesk = {};
+    };
+
+    users.users = mkIf (cfg.user == "tachidesk") {
+      tachidesk = {
+        group = cfg.group;
+        home = cfg.dataDir;
+        description = "Tachidesk Daemon user";
+        isSystemUser = true;
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}/.local/share/Tachidesk' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.tachidesk-server =
+      let
+        #  HOCON is a JSON superset that tachidesk-server use for configuration
+        toHOCON = attr: (
+          let
+            attrType = builtins.typeOf attr;
+          in if builtins.elem attrType [ "string" "path" "int" "float" ] then ''"${toString attr}"''
+            else if attrType == "bool" then lib.boolToString attr
+            # attrs, list, lambda, null
+            else throw ''
+              [tachidesk-server]: invalid config value type '${attrType}'.
+            ''
+        );
+        configFile = pkgs.writeText "server.conf" (lib.strings.concatStrings [
+          ''
+            server.ip = "${cfg.config.ip}"
+            server.port = "${toString cfg.config.port}"
+
+            server.downloadAsCbz = ${lib.boolToString cfg.config.downloadAsCbz}
+            server.localSourcePath = "${cfg.config.localSourcePath}"
+
+            server.basicAuthEnabled = ${lib.boolToString cfg.config.basicAuth.enable}
+          ''
+
+          (
+            if cfg.config.basicAuth.enable then ''
+              server.basicAuthUsername = "${cfg.config.basicAuth.username}"
+              server.basicAuthPassword = "$TACHIDESK_SERVER_BASIC_AUTH_PASSWORD"
+            '' else ""
+          )
+
+          (lib.pipe cfg.extraConfig [
+            (lib.filterAttrs (_: x: x != null))
+            (lib.mapAttrsToList (name: value: ''${name} = ${toHOCON value}''))
+            lib.concatLines
+          ])
+        ]);
+        tmpConfigFile = "/tmp/tachidesk-server.conf";
+      in {
+      description = "A free and open source manga reader server that runs extensions built for Tachiyomi.";
+
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+
+      environment = {
+        TACHIDESK_SERVER_BASIC_AUTH_PASSWORD = if cfg.config.basicAuth.enable then "${lib.fileContents cfg.config.basicAuth.passwordFile}" else "";
+      };
+
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+
+        Type = "simple";
+        Restart = "on-failure";
+        EnvironmentFile = "";
+        ExecStartPre = [
+          "${pkgs.envsubst}/bin/envsubst -i ${configFile} -o ${tmpConfigFile}"
+          "${pkgs.coreutils}/bin/cp ${tmpConfigFile} ${cfg.dataDir}/.local/share/Tachidesk/server.conf"
+          "${pkgs.coreutils}/bin/rm ${tmpConfigFile}"
```

Comment:
Maybe this could just substitute straight into `dataDir`, without needing to go via `/tmp`?

Reply:
I tried but it seemed that envsubst wouldn't work directly by putting `dataDir` directly. I double-check this

Reply:
Well, if it ends up not working and this seems like the best workaround, it would probably warrant a comment. Without any explanation, it seems really unintuitive.

Reply:
Okay it seems to work, so I guess I did something wrong ^^'